src/main/kotlin/no/nav/syfo/aksessering/api/HentFerdigstiltSykmeldingRest.kt

```diff
@@ -0,0 +1,139 @@
+package no.nav.syfo.aksessering.api
+
+import io.ktor.application.call
+import io.ktor.http.HttpStatusCode
+import io.ktor.response.respond
+import io.ktor.routing.Route
+import io.ktor.routing.get
+import io.ktor.routing.route
+import net.logstash.logback.argument.StructuredArguments
+import no.nav.syfo.log
+import no.nav.syfo.model.PapirManuellOppgave
+import no.nav.syfo.model.PapirSmRegistering
+import no.nav.syfo.saf.SafDokumentClient
+import no.nav.syfo.saf.exception.SafForbiddenException
+import no.nav.syfo.service.AuthorizationService
+import no.nav.syfo.service.ManuellOppgaveService
+import no.nav.syfo.util.getAccessTokenFromAuthHeader
+
+fun Route.hentFerdigstiltSykmelding(
+    manuellOppgaveService: ManuellOppgaveService,
+    safDokumentClient: SafDokumentClient,
+    authorizationService: AuthorizationService
+) {
+    route("/api/v1") {
+        get("/sykmelding/{sykmeldingId}/ferdigstilt") {
+            val sykmeldingId = call.parameters["sykmeldingId"]
+
+            log.info("Mottok kall til GET /api/v1/sykmelding/$sykmeldingId/ferdigstilt")
+
+            val accessToken = getAccessTokenFromAuthHeader(call.request)
+            val ferdigstilteOppgaver = sykmeldingId?.let {
+                manuellOppgaveService.hentFerdigstiltManuellOppgave(it)
+            } ?: emptyList()
+            when {
+                accessToken == null -> {
+                    log.info("Mangler JWT Bearer token i HTTP header")
+                    call.respond(HttpStatusCode.Unauthorized)
+                }
+                sykmeldingId == null -> {
+                    log.info("Ugyldig path parameter: sykmeldingId")
+                    call.respond(HttpStatusCode.BadRequest)
+                }
+                ferdigstilteOppgaver.isEmpty() -> {
+                    log.info(
+                        "Fant ingen ferdigstilte manuelloppgaver for sykmledingId {}",
+                        StructuredArguments.keyValue("sykmeldingId", sykmeldingId)
+                    )
+                    call.respond(
+                        HttpStatusCode.NotFound,
+                        "Fant ingen ferdigstilte manuelloppgaver med sykmeldingId $sykmeldingId"
+                    )
+                }
+                else -> {
+                    log.info(
+                        "Henter ut ferdigstilt manuelloppgave med {}",
+                        StructuredArguments.keyValue("sykmeldingId", sykmeldingId)
+                    )
+
+                    if (!ferdigstilteOppgaver.firstOrNull()?.fnr.isNullOrEmpty()) {
+
+                        val manuellOppgave = ferdigstilteOppgaver.first()
+                        val fnr = manuellOppgave.fnr!!
+
+                        if (!authorizationService.hasAccess(accessToken, fnr)) {
```

Comment:
Sjekker du noe sted at veilederen har tilgang ikke bare til personen sykmeldingen gjelder, men også tilgang til å rette en sykmelding?

Reply:
Nei! Godt poeng, det glemte jeg oppi det hele. Todo: kalle på et endepunkt hos iesyfo som sjekker ny ad gruppe for tilgang til å rette sykmelding